src/components/TranslatorList.astro

```diff
@@ -0,0 +1,187 @@
+---
+import { cachedFetch } from '../util-server';
+import { getLanguageFromURL } from '../util';
+
+export interface Props {
+	githubRepo?: `${string}/${string}`;
+}
+
+interface Commit {
+	author: {
+		login: string;
+		id: number;
+	};
+	commit: {
+		message: string;
+	};
+}
+
+interface Contributor {
+	login: string;
+	id: number;
+	contributions: number;
+}
+
+const { githubRepo = 'withastro/docs' } = Astro.props as Props;
+const lang = getLanguageFromURL(Astro.url.pathname);
+
+const printError = (e: Error) =>
+	console.warn(`[error]  /src/components/TranslatorList.astro\n         ${e?.message ?? e}`);
+
+async function getCommitsByPath(path: string, repo: string, page = 1): Promise<Commit[]> {
+	try {
+		const pageSize = 100;
+		const url = `https://api.github.com/repos/${repo}/commits?path=${path}&per_page=${pageSize}&page=${page}`;
+
+		const token = import.meta.env.PUBLIC_GITHUB_TOKEN;
+
+		const res = await cachedFetch(
+			url,
+			{
+				method: 'GET',
+				headers: {
+					Authorization:
+						token && `Basic ${Buffer.from(token, 'binary').toString('base64')}`,
+					'User-Agent': 'astro-docs/1.0',
+				},
+			},
+			{ duration: '15m' }
+		);
+
+		const data = await res.json();
+
+		if (!res.ok) {
+			throw new Error(
+				`Request to fetch commits failed. Reason: ${res.statusText}
+         Message: ${data?.message}`
+			);
+		}
+
+		// Fetch more commits recursively if there are more than GitHub’s per-page response limit.
+		if (data.length === pageSize) {
+			const rest = await getCommitsByPath(path, repo, page + 1);
+			data.push(...rest);
+		}
+
+		return data;
+	} catch (e: any) {
+		printError(e);
+		return new Array();
+	}
+}
+
+function getTranslationContributors(contributions: Commit[]) {
+	const ignoredCommitKeywords = /(en-only|typo|broken link|i18nReady|i18nIgnore)/i;
+	const contributors: Contributor[] = [];
+
+	for (const { author, commit } of contributions) {
+		if (ignoredCommitKeywords.test(commit.message) || !author) continue;
+
+		const contributorIndex = contributors.findIndex(
+			(contributor) => contributor.id === author.id
+		);
+
+		if (contributorIndex < 0) {
+			contributors.push({
+				id: author.id,
+				login: author.login,
+				contributions: 1,
+			});
+			continue;
+		}
+
+		contributors[contributorIndex].contributions += 1;
```

Comment:
Small code style preference nit: we’re checking if a contributor already exists and then using it so we can use `find` over `findIndex` in this case:

```suggestion
		const contributor = contributors.find(
			(contributor) => contributor.id === author.id
		);

		if (!contributor) {
			contributors.push({
				id: author.id,
				login: author.login,
				contributions: 1,
			});
			continue;
		}

		contributor.contributions += 1;
```

(This works because `contributor` is a reference to the object in the `contributors` array, so when we modify a property that’s modified directly on the object in the array.)

An alternative model would be to use a `Map` instead of an array here, so we don’t have to loop over the array each time to find the contributor:

```js
const contributors = new Map<string, Contributor>();

for (const { author, commit } of contributions) {
	const contributor = contributors.get(author.id);
	if (!contributor) {
		contributors.set(author.id, { ... }); 
	}
	contributor.contributions += 1;
}
```